crates/shared/src/oneinch_api.rs

```diff
@@ -512,17 +570,15 @@ mod tests {
     fn swap_query_serialization_options_parameters() {
         let base_url = Url::parse("https://api.1inch.exchange/").unwrap();
         let url = SwapQuery {
-            from_token_address: addr!("EeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE"),
-            to_token_address: addr!("111111111117dc0aa78b770fa6a738034120c302"),
-            amount: 1_000_000_000_000_000_000u128.into(),
             from_address: addr!("00000000219ab540356cBB839Cbe05303d7705Fa"),
             slippage: Slippage::percentage_from_basis_points(50).unwrap(),
-            protocols: Some(vec!["WETH".to_string(), "UNISWAP_V3".to_string()]),
             disable_estimate: Some(true),
-            complexity_level: Some(Amount::new(1).unwrap()),
-            gas_limit: Some(133700),
-            main_route_parts: Some(Amount::new(28).unwrap()),
-            parts: Some(Amount::new(42).unwrap()),
+            common: QuoteAndSwapCommonOptions::with_default_options(
+                addr!("EeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE"),
+                addr!("111111111117dc0aa78b770fa6a738034120c302"),
+                Some(vec!["WETH".to_string(), "UNISWAP_V3".to_string()]),
```

Comment:
Having these protocol options as raw strings seems a bit prone to error. Shouldn't we have an enum somewhere with explicit supported protocols that we use instead?

Reply:
Besides testing (where everything is mocked anyway) we would only ever "instantiate" a protocol ourselves when a user passes a protocol to `solver` or `orderbook` with `--disabled_one_inch_protocols`.
I think introducing an enum for that might be a bit overkill. I fear that it would add too much complexity for the limited benefit it gives us. But asserting that the values passed to `--disabled_one_inch_protocols` are actual protocols and then keep on using the raw strings would already help while being a minimal change.

Reply:
Should be something for another PR, though. To not blow up this one.

Reply:
> Shouldn't we have an enum

My issue with using an `enum` is that these are arbitrarily defined by 1Inch. So if they add a new protocol or change a protocol name, then we need to do a code change (instead of a configuration change) to adjust for it.

Reply:
Sending 1Inch protocols which don't exist, leads to internal server errors.
So I think we could fetch all supported protocols on startup and then either:
1) terminate if we find an unsupported protocol
2) filter out unsupported tokens and print a waning

Reply:
I think this PR will cause major conflicts with #1578. I will merge this PR now to make progress on the other one, but will keep this issue in mind for a future PR.